src/structures/paging/frame_alloc.rs

```diff
@@ -4,8 +4,11 @@ use crate::structures::paging::{PageSize, PhysFrame};
 
 /// A trait for types that can allocate a frame of memory.
 ///
-/// This trait is unsafe to implement because the implementer must guarantee that
-/// the `allocate_frame` method returns only unique unused frames.
+/// # Safety
+///
+/// The implementer of this trait must guarantee that the `allocate_frame`
+/// method returns only unique unused frames. Otherwise, undefined behavior
+/// may result from two caller modifying or deallocating the same frame.
```

Comment:
```suggestion
/// The implementer of this trait must guarantee that the `allocate_frame`
/// method returns only unique unused frames. Otherwise, undefined behavior
/// may result from two callers modifying or deallocating the same frame.
```